ctapipe/image/ellipsoid.py

```diff
@@ -0,0 +1,494 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: UTF-8 -*-
+"""
+Shower image parametrization based on image fitting.
+"""
+
+import astropy.units as u
+import numpy as np
+from astropy.coordinates import Angle
+from iminuit import Minuit
+
+from ctapipe.image.cleaning import dilate
+from ctapipe.image.hillas import hillas_parameters
+from ctapipe.image.leakage import leakage_parameters
+from ctapipe.image.pixel_likelihood import neg_log_likelihood_approx
+from ctapipe.image.toymodel import Gaussian, SkewedGaussian, SkewedLaplace
+
+from ..containers import CameraImageFitParametersContainer, ImageFitParametersContainer
+
+__all__ = ["image_fit_parameters", "ImageFitParameterizationError"]
+
+
+def create_initial_guess(geometry, image, size):
+    """
+    This function computes the seeds of the fit with Hillas parameters
+    Parameters
+    ----------
+    geometry : ctapipe.instrument.CameraGeometry
+        Camera geometry, the cleaning mask should be applied to improve performance
+    image : ndarray
+        Charge in each pixel, the cleaning mask should already be applied to
+        improve performance.
+    size : float/int
+        Total charge after cleaning and dilation
+    Returns
+    -------
+    initial_guess : seed
+    """
+    unit = geometry.pix_x.unit
+    hillas = hillas_parameters(geometry, image)
+
+    initial_guess = {}
+
+    if unit.is_equivalent(u.m):
+        initial_guess["x"] = hillas.x
+        initial_guess["y"] = hillas.y
+    else:
+        initial_guess["x"] = hillas.fov_lon
+        initial_guess["y"] = hillas.fov_lat
+
+    initial_guess["length"] = hillas.length
+    initial_guess["width"] = hillas.width
+    initial_guess["psi"] = hillas.psi
+    initial_guess["skewness"] = hillas.skewness
+
+    if (hillas.width.to_value(unit) == 0) or (hillas.length.to_value(unit) == 0):
+        raise ImageFitParameterizationError("Hillas width and/or length is zero")
+
+    initial_guess["amplitude"] = size
+    return initial_guess
+
+
+def extra_rows(n, cleaned_mask, geometry):
+    """
+    This function adds n extra rows around the cleaned pixels
+    Parameters
+    ----------
+    n : int
+       number of extra rows to add after cleaning
+    cleaned_mask : boolean
+       The cleaning mask applied for Hillas parametrization
+    geometry : ctapipe.instrument.CameraGeometry
+        Camera geometry
+    """
+    mask = cleaned_mask.copy()
+    for row in range(n):
+        mask = dilate(geometry, mask)
+
+    mask = np.array((mask.astype(int) + cleaned_mask.astype(int)), dtype=bool)
+
+    return mask
+
+
+def sensible_boundaries(geometry, cleaned_image, pdf):
+    """
+    Computes boundaries of the fit based on the deviation from Hillas parameters
+    Parameters
+    ----------
+    geometry: ctapipe.instrument.CameraGeometry
+        Camera geometry
+    cleaned_image: ndarray
+        Charge for each pixel, cleaning mask should be applied
+    pdf: str
```

Comment:
Might be useful to use an Enum here and elsewhere for `pdf` (for speed and correctness and to make it easier to include in a config file later).